File: models/Farmer.js
const mongoose = require('mongoose');

const farmerSchema = new mongoose.Schema(
    {
            userId: {
      type: String,
      required: true,
      unique: true,
    },
            name: {
            type: String,
            required: [true, 'Please add a name'],
            trim: true,
                },
            
            phone: {
            type: String,
            required: [true, 'Please add a phone number'],
            },
            
            address: {
            type: String,
            required: [true, 'Please add an address'],
            },
           
            photoUrl: {
            type: String,
            default: '',
            },
            
            qrCodeUrl: {
            type: String,
            default: '',
            },

            isActive:{
           
            type: Boolean,
            default: true,
            },
        },
        {
        timestamps: true,
    }
);


module.exports = mongoose.model('Farmer', farmerSchema)

File: app/src/main/java/model/Farmer.kt
package model


import android.os.Parcelable
import com.google.firebase.firestore.ServerTimestamp
import kotlinx.parcelize.Parcelize
import java.util.Date

@Parcelize
data class Farmer(
    var id: String = "", // Firestore document ID
    val userId: String = "", // KDF-1001, etc.
    val name: String = "",
    val phone: String = "",
    val address: String = "",
    var photoBase64: String? = null, // Photo stored as a Base64 string
    var qrCodeUrl: String? = null,
    val isActive: Boolean = true,
    @ServerTimestamp val createdAt: Date? = null
) : Parcelable